docs/plugins/backend-plugin.md

```diff
@@ -159,37 +227,66 @@ database..
 
 ## Making Use of the User's Identity
 
-The Backstage backend comes with a facility for retrieving the identity of the
-logged in user.
-
-As part of the environment object that is passed to your `createPlugin`
-function, there is a `identity` field. You can use that to get an identity
-from the request.
-
-```ts
-// in packages/backend/src/plugins/carmen.ts
-export default async function createPlugin(
-  env: PluginEnvironment,
-): Promise<Router> {
-  return await createRouter({
-    model: model,
-    logger: env.logger,
-    identity: env.identity,
-  });
+The Backstage backend also offers a core service to access the user's identity. You can access it through the `coreServices.httpAuth` and `coreServices.userInfo` dependencies.
+
+```ts title="plugins/carmen-backend/src/plugin.ts"
+deps: {
+  // highlight-start
+  httpAuth: coreServices.httpAuth,
+  userInfo: coreServices.userInfo
+  // highlight-end
+},
+async init({
+  // highlight-start
+  httpAuth,
+  userInfo,
+  // highlight-end
+}) {
+  httpRouter.use(
+    await createRouter({
+      // highlight-start
+      httpAuth,
+      userInfo,
+      // highlight-end
+      logger,
+    }),
+  );
 }
 ```
 
 The plugin can then extract the identity from the request.
 
 ```ts
+export interface RouterOptions {
+  logger: LoggerService;
+  // highlight-start
+  userInfo: UserInfoService;
+  httpAuth: HttpAuthService;
+  // highlight-end
+}
+
 export async function createRouter(
   options: RouterOptions,
 ): Promise<express.Router> {
-  const router = Router();
-  const { identity } = options;
-
-  router.post('/example', async (req, res) => {
-    const userIdentity = await identity.getIdentity({ request: req });
+  // ...
+  const { userInfo, httpAuth } = options;
+
+  router.post('/me', async (request, response) => {
+    if (!auth.isPrincipal(credentials, 'user')) {
```

Comment:
psst remember to inject auth ;)

Reply:
This particular one is prolly better to do with `httpAuth` instead, as it also throws a standardized error. Pass `allow: ['user']` to the `httpAuth.credentials` just below. Although keep the comment to mention that you generally don't need to include it, it's only if you want to strictly require users

Reply:
getting there... 😁  updated in [f50d970](https://github.com/backstage/backstage/pull/24069/commits/f50d970af685f490c3367f74d817269159f29268)

Reply:
ah, @Rugvip can you make the change? I haven't used the new auth services, so not terribly familiar here.

Reply:
ah good point how could i forget